src/controller/SetUpCodePairer.h

```diff
@@ -52,12 +54,13 @@ class SetUpCodePairerParameters : public RendezvousParameters
 {
 public:
     SetUpCodePairerParameters() = default;
-    SetUpCodePairerParameters(const Dnssd::CommonResolutionData & data, size_t index);
+    SetUpCodePairerParameters(const Dnssd::CommonResolutionData & data, std::optional<uint16_t> longDiscriminator, size_t index);
 #if CONFIG_NETWORK_LAYER_BLE
-    SetUpCodePairerParameters(BLE_CONNECTION_OBJECT connObj, bool connected = true);
+    SetUpCodePairerParameters(BLE_CONNECTION_OBJECT connObj, std::optional<uint16_t> longDiscriminator, bool connected = true);
```

Comment:
if you switch the order of parameters, it would be more backwards compatible with old code ... or we are trying to fail at compile time on purpose?